session/objecthandling.go

```diff
@@ -706,44 +706,60 @@ func checkHyperCubeErrors(id string, hypercube *enigmahandlers.HyperCube, logEnt
 		return errors.Errorf("object<%s> has no hypercube", id)
 	}
 
-	if err := checkHyperCubeErrorInner(id, "hypercube", hypercube.Error); err != nil {
+	if err, _ := checkHyperCubeErrorInner(id, "hypercube", hypercube.Error); err != nil {
 		return err
 	}
 
+	var firstWarnError error
 	if hypercube.DimensionInfo != nil {
 		for i, dimInfo := range hypercube.DimensionInfo {
-			if err := checkHyperCubeErrorInner(id, fmt.Sprintf("hypercube.DimensionInfo[%d]", i), dimInfo.Error); err != nil {
-				logEntry.Log(logger.WarningLevel, fmt.Sprintf("object<%s> has hypercube error<%s> in DimensionInfo[%d]", id, EngineCodeToString(dimInfo.Error.ErrorCode), i))
+			if err, warning := checkHyperCubeErrorInner(id, fmt.Sprintf("hypercube.DimensionInfo[%d]", i), dimInfo.Error); err != nil {
+				if !warning {
+					return errors.Wrapf(err, "object<%s> has hypercube error<%s> in DimensionInfo[%d]", id, EngineCodeToString(dimInfo.Error.ErrorCode), i)
+				}
+				if firstWarnError != nil {
+					firstWarnError = err
```

Comment:
Can't we simply return not return here and in the other similar if statements?

Reply:
the other non warn errors would not be found if there is an "error" which should only be warning

Reply:
Right I get it!

Reply:
yepp, I'll fix

Reply:
fixed